Classes/Event/EventPublisher.php

```diff
@@ -105,7 +106,10 @@ public function publishMany(string $streamName, array $events, int $expectedVers
         foreach ($rawEvents as $rawEvent) {
             $eventClassName = $this->eventTypeResolver->getEventClassNameByType($rawEvent->getType());
             $event = $this->propertyMapper->convert($rawEvent->getPayload(), $eventClassName, $configuration);
-            foreach ($this->eventListenerLocator->getSynchronousListeners($rawEvent->getType()) as $listener) {
+            foreach ($this->eventListenerLocator->getSynchronousListenersByEventType($rawEvent->getType()) as $listener) {
+                if ($listener[0] instanceof ActsBeforeInvokingEventListenerMethodsInterface) {
+                    $listener[0]->beforeInvokingEventListenerMethod($listener[1], $event, $rawEvent);
```

Comment:
yeah, I read it … I just wondered: you (we all) are using it as an array elsewhere already, so that would have to be changed, too, right? So should we check if it's an array and if it's not, ignore it in the relevant parts? 

Actually, I think it's YAGNI if nobody needed it **yet**. If we don't have a case for it right now, I'd rather remove it and add support for closures later again. As far as I can see it doesn't have an impact on userland anyway, right?